eng/Subsets.props

```diff
@@ -218,15 +220,30 @@
   </ItemGroup>
 
   <ItemGroup Condition="$(_subsetCategory.Contains('installer')) and $(_subset.Contains('-depproj-'))">
-    <DepprojProjectToBuild Include="$(InstallerProjectRoot)pkg\projects\**\*.depproj" SignPhase="R2RBinaries" BuildInParallel="false" />
+    <DepprojProjectToBuild Condition="'$(RuntimeFlavor)' == 'CoreCLR'" Include="$(InstallerProjectRoot)pkg\projects\**\*.depproj" SignPhase="R2RBinaries" BuildInParallel="false" />
+    <DepprojProjectToBuild Condition="'$(RuntimeFlavor)' == 'Mono'" Include="$(InstallerProjectRoot)pkg\projects\**\*.depproj" SignPhase="Binaries" BuildInParallel="false" />
     <ProjectToBuild Include="@(DepprojProjectToBuild)" />
   </ItemGroup>
 
-  <ItemGroup Condition="$(_subsetCategory.Contains('installer')) and $(_subset.Contains('-pkgproj-'))">
+  <ItemGroup Condition="$(_subsetCategory.Contains('installer')) and $(_subset.Contains('-pkgproj-')) and '$(TargetOS)' != 'iOS'">
     <PkgprojProjectToBuild Include="$(InstallerProjectRoot)pkg\projects\**\*.pkgproj" SignPhase="MsiFiles" BuildInParallel="false" />
     <ProjectToBuild Include="@(PkgprojProjectToBuild)" />
   </ItemGroup>
 
+  <ItemGroup Condition="$(_subsetCategory.Contains('installer')) and $(_subset.Contains('-pkgproj-')) and '$(TargetOS)' == 'iOS'">    
```

Comment:
My initial impression is that I don't like seeing this complexity in the subsets system. It seems more maintainable and readable (discoverable) to me for the pkgproj/depproj/etc. to be able to do the right thing when encountering this new TargetOS.

Reply:
For example, pkgproj Build has functionality to allow a complete skip:

https://github.com/dotnet/arcade/blob/7c5a68c1aeafec3a9a1b63d1d2c3b197964cc1b7/src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.packaging.targets#L237-L239

A potential way is to manipulate `BuildOnUnknownPlatforms` to trigger a skip based on RID not being present:

https://github.com/dotnet/runtime/blob/1d018ba7e9bfcbc8eb1f56fd14532e7fcfc480c1/src/installer/pkg/projects/Directory.Build.props#L117-L119

https://github.com/dotnet/arcade/blob/866b2acd5ffc3c2031c102f6415fd7c6a1a370d5/src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.targets#L167-L168

Reply:
So, I skipped at the pkgproj level.  It worked, but it still generated an empty nupkg for each project.  It's unclear if I can actually skip that step.  Do you know?

Reply:
Yes, disabling `BuildOnUnknownPlatforms` prevents the WindowsDesktop packages from building where they shouldn't in Core-Setup release/3.1. Wouldn't expect to see empty nupkgs in any situation I know about.